src/prpy/util.py

```diff
@@ -1598,3 +1598,35 @@ def GetManipulatorIndex(robot, manip=None):
             manip_idx = manip.GetRobot().GetActiveManipulatorIndex()
 
     return (manip, manip_idx)
+
+def GetPointFrom(env, focus):
+    """
+    Given a kinbody, array or transform, returns the xyz
+    location. 
+    param env The environment where the item exists
+    param focus THe area to be referred to
+    """
+    #Pointing at an object
+    if isinstance(focus, (openravepy.KinBody, openravepy.KinBody.Link)):
+        with env:
```

Comment:
Actually, that's not entirely true. It would be `focus.GetEnv()` for `KinBody` and `focus.GetParent().GetEnv()` for `Link`. But my point still stands.